lib/logstash/inputs/udp.rb

```diff
@@ -74,11 +72,13 @@ def udp_listener(output_queue)
       Thread.new { inputworker(i) }
     end
 
-    while true
+    while !stop?
       #collect datagram message and add to queue
       payload, client = @udp.recvfrom(@buffer_size)
```

Comment:
I think being non blocking is not that hard to introduce, there is a clean api with error code check for that, so why not adding it too? don't you think? 

I for example did it in some of my PR where I found blocking calls (example the zeromq), this small introduction would about having to actually close the recv throw the exception close .... but at the end of the day is a choice, so no objections if you don't want to do it here.